src/lib/crypto/kimchi_backend/common/plonk_constraint_system.ml

```diff
@@ -951,8 +967,60 @@ end = struct
                 let value = compute lc in
                 res.(col_idx).(row_idx) <- value ;
                 Hashtbl.set internal_values ~key:var ~data:value ) ) ;
+
+    let map_runtime_tables = MapRuntimeTable.Table.create () in
+    let runtime_tables : Fp.t Kimchi_types.runtime_table array =
+      match sys.runtime_tables_cfg with
+      | Unfinalized_runtime_tables_cfg_rev _ ->
+          failwith
+            "Attempted to generate a witness for an unfinalized constraint \
+             system"
+      | Compiled_runtime_tables_cfg cfgs ->
+          Array.mapi cfgs ~f:(fun rt_idx { Kimchi_types.id; first_column } ->
+              let data =
+                Array.mapi first_column ~f:(fun i v ->
+                    ignore
+                    (* `add` leaves the value unchanged if the index has been
```

Comment:
TODO, non-urgent: a user may want to duplicate the 'index' values, and have different values associated with the indices. We should be able to handle this, by overwriting the corresponding entries as we fill them up.
E.g. if you have a runtime table with indices `[1, 1]`, you should be able to lookup `[(1, 1), (1, 2)]`

Reply:
Opened the follow-up issue: https://github.com/MinaProtocol/mina/issues/14016